src/main/java/org/folio/innreach/domain/service/impl/CirculationServiceImpl.java

```diff
@@ -262,6 +267,28 @@ public InnReachResponseDTO returnUncirculated(String trackingId, String centralC
     }
   }
 
+  @Override
+  public InnReachResponseDTO borrowerRenew(String trackingId, String centralCode, BorrowerRenewDTO borrowerRenew) {
+    var transaction = getTransaction(trackingId, centralCode);
+    var calculatedDueDate = transaction.getHold().getNeedBefore();
+    var requestedDueDate = borrowerRenew.getDueDateTime();
+
+    if (calculatedDueDate >= requestedDueDate) {
+      transaction.setState(BORROWER_RENEW);
+      transactionRepository.save(transaction);
+    } else {
+      try {
+        var recallItem = new InnReachRecallItemDTO().dueDateTime(calculatedDueDate);
+        transactionService.recallItem(trackingId, centralCode, recallItem);
+      } catch (IllegalArgumentException e) {
+        var cancelRequest = new CancelRequestDTO().reason("Transaction cancelled at owning side");
+        cancelPatronHold(trackingId, centralCode, cancelRequest);
+      }
```

Comment:
Is there a requirement to cancel transaction? I guess it's better to clarify with PO. 
Also, pay attention to the type of hold. The method `cancelPatronHold` is designed specifically for transactions of type `TransactionType.Patron` while `Item Hold` is of type `TransactionType.Item`

Reply:
removed